### PR TITLE
fix: make GetCatalogsBuilder sort catalog names 

### LINKED ISSUE #6807: [FlightSQL] GetCatalogsBuilder does not sort the catalog names
The GetCatalogsBuilder does not apply sorting as specified in the documentation comments. In contrast, other builders, such as GetDbSchemasBuilder and GetTablesBuilder, sort the record batch exactly as described in their documentation.

See: https://github.com/apache/arrow-rs/blob/e1e881405ce9b180858d5ba278803b4795b63e74/arrow-flight/src/sql/arrow.flight.protocol.sql.rs#L120-L122

And see: https://github.com/apache/arrow-rs/blob/e1e881405ce9b180858d5ba278803b4795b63e74/arrow-flight/src/sql/metadata/catalogs.rs#L70-L79

### PR DESCRIPTION
**Pull request by niebayes**
# Which issue does this PR close?

Closes #6807.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

No.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @niebayes

File: arrow-flight/src/sql/metadata/catalogs.rs [hunk 1]
```diff
@@ -68,7 +68,8 @@ impl GetCatalogsBuilder {
     /// builds a `RecordBatch` with the correct schema for a
     /// [`CommandGetCatalogs`] response
     pub fn build(self) -> Result<RecordBatch> {
-        let Self { catalogs } = self;
+        let Self { mut catalogs } = self;
+        catalogs.sort();
```
Comment (tustvold) [inline arrow-flight/src/sql/metadata/catalogs.rs:72]:
```suggestion
        catalogs.sort_unstable();
```

**Review by tustvold — APPROVED** [review]
Just a minor nit

File: arrow-flight/src/sql/metadata/catalogs.rs (hunk 1 above)
Comment (niebayes) [reply arrow-flight/src/sql/metadata/catalogs.rs:72]: @tustvold Thank you!

**Comment by alamb** [comment]
Thanks @niebayes  and @tustvold

**Pull request merged**